_posts/2025-02-03-Reduce-Cost-with-Disk-based-Vector-Search.md

```diff
@@ -0,0 +1,264 @@
+
+| Dataset      | [MS Marco 2.1 encoded with the Cohere v3 model](https://huggingface.co/datasets/Cohere/msmarco-v2.1-embed-english-v3)                                       |
+| Dimension  | 1024                                                                     |
+| Normalized      | Yes                                                                      |
+| Space type | Cosine (inner product over normalized data)                              |
+| Index vectors | 113M                                                                     |
+
+We tested four different cluster configurations using [opensearch-cluster-cdk](https://github.com/opensearch-project/opensearch-cluster-cdk) with OpenSearch 2.18. We selected the cluster and index configuration to follow production recommendations. For example, we configured replica shards and dedicated cluster manager nodes. In addition, we targeted a shard count that provides 2 to 3 vCPUs per shard. The following table presents the configurations for these tests.
+
+| Name        | Data node count | Data node type | Data node disk size | Data node disk type | JVM size | Primary shard count | Replica shards | Compression level |
+|-------------|-----------------|----------------|---------------------|---------------------|----------|---------------------|----------------|-------------------|
+| `in_memory`   | 8               | r6g.8xlarge    | 300                 | EBS                 | 32       | 40                  | 1              | 1x                |
+| `on_disk_8x`  | 10              | r6gd.2xlarge   | 474                 | Instance            | 32       | 15                  | 1              | 8x                |
+| `on_disk_16x` | 6               | r6gd.2xlarge   | 474                 | Instance            | 32 | 9    | 1                   | 16x            |
+| `on_disk_32x` | 4               | r6gd.2xlarge   | 474                 | Instance            | 32 | 6    | 1                   | 32x            |
+
+In the table, note that clusters with higher compression levels use significantly less resources than those without compression.
+
+To optimize performance for our tests, we made these additional adjustments:
+
+- Retrieved IDs from `doc_values` to reduce fetch time
+- Disabled `_source` storage 
+
+We ran all tests using OpenSearch Benchmark [vector search workloads](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/vectorsearch), following this procedure:
+
+1. Ingest the dataset.
+2. Force merge the index to 5 segments per shard.
+3. Run warm-up queries to load the index into memory.
+4. Run the single-client latency test.
+5. Run the multi-client throughput tests.
```

Comment:
Confirm that singular "test" on line 240 but plural "tests" on line 241 is correct.

Reply:
should both be test